### ISSUE TITLE
Wikimedia Two-Factor Authentication (2FA) in OpenRefine

### ISSUE THREAD

**Issue by jinoytommanjaly** [OpenRefine/OpenRefine#3231]
Wikimedia has implementation two-factor authentication (2FA) is a way to strengthen the security of your account. So users who has enabled two-factor authentication will be asked for a one-time six-digit authentication code every time in addition to the password. Currently those who has enabled 2FA can't login to Wikidata in OpenRefine.

Right now Two-factor authentication on Wikimedia is a experimental and optional. And enrollment requires (oathauth-enable) access account. which is currently for account with administrators, bureaucrats, checkusers, oversighters, stewards, edit filter managers and the OATH-testers global group.

### Proposed solution
So i the account holder who has enabled Two-factor authentication will not be able to login to Wikidata from OpenRefine. He/She has to disable 2FA to login using OpenRefine.

So Enable 2FA key input option after entering the username and password in future updates

**Comment by jinoytommanjaly** [comment]
Duplicate of #1612

**Comment by wetneb** [comment]
At the moment we are considering adding wording to encourage people to use bot passwords: #3108. Do you know if bot passwords can still be used when 2FA is enabled?

For proper 2FA support we would need that to be supported by Wikidata-Toolkit: https://github.com/Wikidata/Wikidata-Toolkit/issues/412

**Comment by jinoytommanjaly** [comment]
@wetneb Yes I have checked right now. Bot passwords can be used for login when 2FA is enabled.

**Comment by wetneb** [comment]
Thanks! So I will leave this open since the request is still legitimate, but hopefully #3108 should be a satisfactory short-term mitigation.
